lambda_functions/two_way_sms/two_way_sms_v2.py

```diff
@@ -0,0 +1,344 @@
+        except (json.decoder.JSONDecodeError, TypeError) as e:
+            # Deadletter
+            push_to_sqs(record_body, False)
+            logger.error("Malformed record body.")
+            logger.exception(e)
+            continue
+
+        if not valid_message_body(inbound_sms):
+            logger.critical("The record's message body is invalid: %s", inbound_sms)
+
+            # Push the record to the dead letter queue, and return 200 to have the message removed from feeder queue.
+            push_to_sqs(record_body, False)
+            return create_response(200)
+
+        # destinationNumber is the number the end user responded to (the 10DLC pinpoint number).
+        two_way_record = two_way_sms_table_dict.get(inbound_sms.get("destinationNumber"))
+        if two_way_record is None:
+            # Deadletter
+            push_to_sqs(record_body, False)
+            logger.critical("Unable to find two_way_record for: %s", inbound_sms.get("destinationNumber", "unknown"))
+            continue
+
+        # **Note** - Commenting out this code for self managed checking because right now we are relying on AWS to manage opt out/in functionality.  
+        # **Note** - Eventually we will migrate to self-managed for everything and the config determination will be whether Notify is handling the functionality or if the business line is
+        # If the number is not self-managed, look for key words
+        #if not two_way_record.get('self_managed'):
+        #    logger.info('Service is not self-managed')
+        #    keyword_phrase = detected_keyword(inbound_sms.get('messageBody', ''))
+        #    if keyword_phrase:
+        #        # originationNumber is the veteran number.
+        #        send_message(two_way_record.get('originationNumber', ''),
+        #                     two_way_record.get('destinationNumber', ''),
+        #                     keyword_phrase)
+
+        # Forward inbound_sms to the associated service.
+        logger.info(
+            "Forwarding inbound SMS to service: %s. UrlEndpoint: %s",
+            two_way_record.get("service_id"),
+            two_way_record.get("url_endpoint")
+        )
+
+        try:
+            result_of_forwarding = forward_to_service(inbound_sms, two_way_record.get("url_endpoint"))
+        except Exception as e:
+            # Deadletter
+            push_to_sqs(record_body, False)
+            logger.exception(e)
+            continue
+
+        if not result_of_forwarding:
+            logger.info("Failed to make an HTTP request.  Placing the request back on retry.")
+            # put back on replay queue
+            push_to_sqs(record_body, True)
+            batch_item_failures.append({"itemIdentifier": record_body.get("messageId", '')})
+
+    # Return an array of message Ids that failed so that they get re-enqueued.
+    return {
+        "data": batch_item_failures,
+        "statusCode": 400 if batch_item_failures else 200,
+    }
```

Comment:
Has this functionality been tested? I just learned it's a thing today.

Reply:
To clarify, returning a list of the Ids that need to be re-enqueued.

Reply:
Whatever we return here wont matter because if we return anything it'll be considered successful.  The only thing that would make it unsuccessful is if we throw an error.  Throwing an error would replay the message after visibility timeout expires.  

When we put in batching, we can return just the batch_item_failures array and it will replay only the message ids that are in the array.

Reply:
A unit test failed when I only returned `batch_item_failures`.  It passes with this return type.  When we put in batching, we can restore the return value and update the unit test.